From 71e8e543b1abee2f9c7921f7ef8e0d9fba3e3ee3 Mon Sep 17 00:00:00 2001
From: Johanan Lai <lai.johanan@gmail.com>
Date: Fri, 29 Jul 2016 10:06:13 -0700
Subject: [PATCH] ADMINUI-2321 Data center should be immutable when editing
 user provisioning limits ADMINUI-2326 Clicking "New Limit" twice while
 provisioning a limit causes the form to change from New Limit -> Update Limit
 for undefined

---
 www/js/components/pages/user/limits/form.jsx | 4 ++--
 www/js/components/pages/user/limits/main.jsx | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/www/js/components/pages/user/limits/form.jsx b/www/js/components/pages/user/limits/form.jsx
index 47a01ee2..15194cf8 100644
--- a/www/js/components/pages/user/limits/form.jsx
+++ b/www/js/components/pages/user/limits/form.jsx
@@ -172,7 +172,7 @@ var LimitsForm = React.createClass({
                     <div className="form-group">
                         <label className="col-md-4 control-label">Datacenter</label>
                         <div className="col-md-7">
-                            <Chosen className="form-control" placeholder="Select a Datacenter" value={this.state.datacenter} onChange={this.handleChangeDatacenter}>
+                            <Chosen className="form-control" placeholder="Select a Datacenter" value={this.state.datacenter} onChange={this.handleChangeDatacenter} disabled={this.props.initialLimit}>
                             <option value=""></option>
                             {
                                 this.state.datacenters.map(function (dc) {
@@ -233,7 +233,7 @@ var LimitsForm = React.createClass({
 
                     <div className="form-group">
                         <div className="input-group col-sm-offset-4 col-sm-6">
-                            <button disabled={(this.state.datacenter && this.isValid()) ? '' : 'disabled'} onClick={this.handleSave} className="btn btn-primary" type="submit">Save Limit</button>
+                            <button disabled={!(this.state.datacenter && this.isValid())} onClick={this.handleSave} className="btn btn-primary" type="submit">Save Limit</button>
                             <button type="button" onClick={this.props.handleCancel} className="btn btn-default">Cancel</button>
                         </div>
                     </div>
diff --git a/www/js/components/pages/user/limits/main.jsx b/www/js/components/pages/user/limits/main.jsx
index 06c1b3ef..94a69d04 100644
--- a/www/js/components/pages/user/limits/main.jsx
+++ b/www/js/components/pages/user/limits/main.jsx
@@ -90,7 +90,7 @@ var ProvisioningLimits = React.createClass({
                 <h3>Provisioning Limits
                     {adminui.user.role('operators') &&
                         <div className="actions">
-                            <button onClick={this.showNewLimitForm} className="btn btn-sm btn-info"><i className="fa fa-plus"></i> New Limit</button>
+                            <button onClick={this.showNewLimitForm} className="btn btn-sm btn-info" disabled={this.state.form}><i className="fa fa-plus"></i> New Limit</button>
                         </div>}
                 </h3>
                 <ProvisioningLimitsList
-- 
2.21.0

